open-api/rest-catalog-open-api.yaml

```diff
@@ -2106,6 +2210,32 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        An opaque JSON object that contains information provided by the REST server
+        to be utilized by clients for distributed table scan planning; should be supplied
+        as is for input in `PlanTable` operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/DataFile'
+        delete-files:
+          oneOf:
```

Comment:
Is `oneOf` correct here?  A dataFile could have both an equality and positional delete file that applies.

Reply:
@danielcweeks @rdblue  So to confirm a list of `delete-files` can have a mix of `positional` deleteFile and `equality` deleteFiles? I thought it can only be a list only of positional deletes, or a list of equality deletes

When checking the java interface for just the `DeleteFile` https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/DeleteFile.java#L24 which is based on `ContentFile`, it seems a single file can only have one type https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/ContentFile.java#L42 
```
one of DATA, POSITION_DELETES, or EQUALITY_DELETES
```

Reply:
The list in this response representation is the list of delete files that need to be applied for the given data file in a given task.

That list can have both position and equality deletes. The java representation being referred to is just for a single instance of a delete file, but certainly there can be mutiple position + equality deletes that need to be applied to a single data file in a task. I'd expect the task to have both of those since it's required for a correct reading of a table.

E.g. There can be Data File A, and then Position Delete File  which deletes a row in Data File A, and then also equality delete can be applied which is relevant for some rows in Data File A. The FileScanTask should have data file 1 be the data file, and a List containing both the position delete and equality delete.

So yeah I think oneOf is not correct here; we'll either want to 

1.) see if there's some other construct to help us define a list where the entries can be a mix of both data and delete files

Or 

2.) Define separate lists for equality and position deletes. 

I'd prefer 1 but I'm not really familiar enough yet with the open API syntax to define that; worth looking into that first though.

Reply:
I have changed it to 2 separated lists of `equality-delete-files` and `position-delete-files`. Please take a look.

Reply:
I think it's still marked as `oneOf` and it looks like it enforces just one list or the other.  Should this be `anyOf`?

Reply:
It seems like we might be able define this as:

```yaml
        delete-files:
          type: array
          items:
            anyOf:
              - $ref: '#/components/schemas/PositionDeleteFile'
              - $ref: '#/components/schemas/EqualityDeleteFile'
```

I think each have a type enum that can be used to determine the underlying file type.  That way there's just one list of delete files.